sqlacodegen/codegen.py

```diff
@@ -683,6 +685,26 @@ def _get_kwargs_repr(kwargs):
             if isinstance(relationship, Relationship):
                 rendered += '{0}{1} = {2}\n'.format(self.indentation, attr, self.render_relationship(relationship))
 
+                 # Workaround for relationships of depth 2 between User class and Role class
+                # For example, if the flask roles are stored in a 'Privlages' table,
+                # and the user is associated with a 'User -> Positions -> Privlages' relationship,
+                # then the roles class is 'Privlages' and the linking class is 'Positions'
+                classes = filter(lambda model: isinstance(model, self.class_model), self.models)
+                class_dict = {c.name: c for c in classes}
+                role_name = self.flask_login_role.lower()
+                link_name = relationship.target_cls
+                # If we need a role relationship and this user does not have a role collumn
+                #       AND this relationship is the one connected to the table with the roles relationship
+                if model.name == self.flask_login_user and self.flask_login_role and self.flask_login_role != 'Role' \
+                    and role_name+'s' in class_dict[link_name].attributes:
+
+                    link_name = link_name.lower()
+                    user_link = f"t_{model.name.lower()}_{link_name}"
+                    link_role = f"t_{link_name}_{role_name}"
+                    join = f"join({user_link}, {link_role}, {user_link}.c.{link_name}_id == {link_role}.c.{link_name}_id)"
```

Comment:
It may not look like much, but the functionality in this line took me 5 hours, and was originally 2 files. I am so glad its in the form it is today.

Reply:
Nice work!